test/tests/auxkernels/array_parsed_aux/array_parsed_aux.i

```diff
@@ -0,0 +1,96 @@
+[Mesh]
+  [gmg]
+    type = GeneratedMeshGenerator
+    dim = 1
+    nx = 10
+  []
+[]
+
+[Variables]
+  [u]
+    components = 2
+  []
+  [v]
+    components = 2
+  []
+[]
+
+[AuxVariables]
+  [const]
+    initial_condition = 0.5
+  []
+  [parsed]
+    components = 2
+  []
+  [sum]
+  []
+[]
+
+[Functions]
+  [cosx]
+    type = ParsedFunction
+    expression = 'cos(x)'
+  []
+  [sinx]
+    type = ParsedFunction
+    expression = 'sin(x)'
+  []
+  [px]
+    type = ParsedFunction
+    expression = 'x'
+  []
+  [mx]
+    type = ParsedFunction
+    expression = '-x'
+  []
+[]
+
+[ICs]
+  [uic]
+    type = ArrayFunctionIC
+    variable = u
+    function = 'cosx sinx'
+  []
+  [vic]
+    type = ArrayFunctionIC
+    variable = v
+    function = 'px mx'
+  []
+[]
+
+[AuxKernels]
+  [parsed_aux]
+    type = ArrayParsedAux
+    variable = parsed
+    expression = '(u^2 + v)*(x - const)*factor'
+    coupled_array_variables = 'u v'
+    coupled_variables = const
+    constant_names = 'factor'
+    constant_expressions = '3.14'
+    use_xyzt = true
```

Comment:
I'm wondering whether this should be the default?

Reply:
I'm willing to change, but should we stray from `ParsedAux` and `ParsedVectorAux` syntax?

Reply:
I suppose not in this PR. We can update them uniformly later if we want